src/libcore/num/mod.rs

```diff
@@ -4496,6 +4518,19 @@ pub use num::dec2flt::ParseFloatError;
 // Conversions T -> T are covered by a blanket impl and therefore excluded
 // Some conversions from and to usize/isize are not implemented due to portability concerns
 macro_rules! impl_from {
+    (bool, $Large: ty, #[$attr:meta]) => {
+        #[$attr]
+        impl From<bool> for $Large {
+            #[inline]
+            fn from(small: bool) -> $Large {
```

Comment:
This can just use an `as` cast.

Reply:
`impl From` will be removed as #50554 gets merged.